napari/layers/utils/layer_utils.py

```diff
@@ -223,15 +336,52 @@ def dataframe_to_properties(
     dict[str, np.ndarray]
         A properties dictionary where the key is the property name and the value
         is an ndarray with the property value for each point.
-    Optional[dict[int, int]]
-        mapping from label number to position (row) in properties
     """
+    return {col: np.asarray(dataframe[col]) for col in dataframe}
+
+
+def validate_properties(
```

Comment:
To at least partially address the comment above, inside `validate_properties` we can maybe coerce dtypes to anything more specific than `object`, if possible.

Reply:
My understanding is that numpy should mostly handle that for us. The only way a property value array should have `dtype=object` is either because that was explicitly defined on construction, an array-like contains mixed types, or an array-like contains non-string/numeric data.

Agreed that we should aim to handle any unsupported cases gracefully or with useful error messages, but is there a reason to think that we should re-coerce data types?

Reply:
I think the issues happened for me with an array of strings only that somehow had object as dtype. But maybe as you said this won't be a problem now; let's come back to it if it shows up again!